src/ape/managers/query.py

```diff
@@ -76,12 +78,47 @@ def perform_block_transaction_query(
         return self.provider.get_transactions_by_block(query.block_id)
 
     @perform_query.register
-    def perform_contract_creation_query(self, query: ContractCreationQuery) -> Iterator[ReceiptAPI]:
-        yield from self.provider.get_contract_creation_receipts(
-            address=query.contract,
-            start_block=query.start_block,
-            stop_block=query.stop_block,
-        )
+    def perform_contract_creation_query(
+        self, query: ContractCreationQuery
+    ) -> Iterator[ContractCreation]:
+        def find_creation_block(lo, hi):
+            # perform a binary search to find where the contract code appeared
+            # takes log2(height), doesn't work with contracts that have been reinit.
+            while hi - lo > 1:
+                mid = (lo + hi) // 2
+                code = self.provider.get_code(query.contract, block_id=mid)
+                if not code:
+                    lo = mid
+                else:
+                    hi = mid
+
+            if self.provider.get_code(query.contract, block_id=hi):
+                return hi
+
+        # if it doesn't have code at head, skip the search
+        if not self.provider.get_code(query.contract):
+            return None
+
+        block = find_creation_block(0, self.chain_manager.blocks.height)
```

Comment:
block may be None

Reply:
when?

Reply:
If this raised an exception it would not longer return None in some rare future scenario